src/main/java/com/answerdigital/answerking/service/TagService.java

```diff
@@ -118,4 +120,27 @@ private void addProductsToTag(final Tag tag, final List<Long> productIds) {
         productService.validateProductsAreNotRetired(products);
         tag.setProducts(products);
     }
+
+    /**
+     * Retires a Tag {@link com.answerdigital.answerking.model.Tag}.
+     * @param tagId The Tag ID to retire.
+     */
+    public void retireTag(final Long tagId) {
+        final Tag tag = findById(tagId);
+        validateTagIsNotRetired(tag);
+        tag.setRetired(true);
+        tagRepository.save(tag);
+    }
+
+    /**
+     * Checks if a tag is retired {@link com.answerdigital.answerking.model.Tag}
+     * @param tag the Tag to validate the retirement of {@link com.answerdigital.answerking.model.Tag}.
+     * @throws RetirementException When a Tag {@link com.answerdigital.answerking.model.Tag} is retired.
+     */
+    private void validateTagIsNotRetired(final Tag tag) {
```

Comment:
Could we please address the TODO comment on 'updateTag' and verify that the tag is not retired when updating?